redwood-flexbox-engine/src/commonMain/kotlin/app/cash/redwood/utils.kt

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright (C) 2022 Square, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package app.cash.redwood
+
+import app.cash.redwood.Node.Companion.MatchParent
+import app.cash.redwood.Node.Companion.WrapContent
+import kotlin.math.max
+import kotlin.math.min
+
+internal const val Undefined = -1
+
+internal fun packLong(lower: Int, higher: Int): Long {
+  return (higher.toLong() shl 32) or (lower.toLong() and 0xFFFFFFFF)
```

Comment:
supernit: nicer to do a trailing `L` here, like `0xFFFFFFFFL`. It’s gonna be a long in bytecode and the only reason you get that for free is 'cause it’s bigger than max int.